src/extension/importer/core/src/main/java/org/geoserver/importer/transform/AttributeComputeTransform.java

```diff
@@ -0,0 +1,87 @@
+/* (c) 2018 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.importer.transform;
+
+import org.geoserver.importer.ImportTask;
+import org.geotools.data.DataStore;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.filter.expression.Expression;
+
+/**
+ * Transform creating a new attribute based on the existing ones
+ */
+public class AttributeComputeTransform extends AbstractTransform implements InlineVectorTransform {
+    
+    private static final long serialVersionUID = 1L;
+
+    /** field to remap */
+    protected String field;
+
+    /** type to remap to */
+    protected Class type;
+    
+    /** the expression to apply (stored as a string as CQL does not always round trip properly */
+    protected String cql;
+    protected Expression expression;
+    
+    public AttributeComputeTransform(String field, Class type, String cql) throws CQLException {
+        this.field = field;
+        this.type = type;
+        this.cql = cql;
+        this.expression = ECQL.toExpression(cql);
```

Comment:
I would use ``setCql(...)`` method here since this is not a single line assignment. This will remove the risk of editing only the behavior of the ``setCql(...)`` or vice versa.

Reply:
Roger